types/address_test.go

```diff
@@ -507,15 +565,234 @@ func (s *addressTestSuite) TestGetFromBech32() {
+	addr := &types.ValAddress{}
+	err := addr.UnmarshalJSON([]byte{34, 34})
+	s.Require().NoError(err)
+	s.Require().Equal(&types.ValAddress{}, addr)
+}
+
+func (s *addressTestSuite) TestUnmarshalYAMLValAddressFailed() {
+	malformedYAML := []byte("k:k:K:")
+	addr := &types.ValAddress{}
+	err := addr.UnmarshalYAML(malformedYAML)
+	s.Require().Error(err)
+}
+
+func (s *addressTestSuite) TestUnmarshalYAMLValAddressWithEmptyString() {
+	addr := &types.ValAddress{}
+	err := addr.UnmarshalYAML([]byte{34, 34})
+	s.Require().NoError(err)
+	s.Require().Equal(&types.ValAddress{}, addr)
+}
+
+func (s *addressTestSuite) TestFormatValAddressAsString() {
+	addr20byte := []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19}
+	accAddr := types.ValAddress(addr20byte)
+
+	actual := fmt.Sprintf("%s", accAddr)
+
+	hrp := types.GetConfig().GetBech32ValidatorAddrPrefix()
+	expected, err := bech32.ConvertAndEncode(hrp, addr20byte)
+	s.Require().NoError(err)
+	s.Require().Equal(expected, actual)
+}
+
+func (s *addressTestSuite) TestFormatValAddressAsPointer() {
+	accAddr := types.AccAddress([]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+	ptrAddr := &accAddr
+	actual := fmt.Sprintf("%p", ptrAddr)
+	expected := uintptr(unsafe.Pointer(&accAddr))
+	s.Require().Equal(fmt.Sprintf("0x%x", expected), actual)
+}
+
+func (s *addressTestSuite) TestFormatValAddressWhenVerbIsDifferentFromSOrP() {
+	myAddr := types.ValAddress([]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+	exp := "000102030405060708090A0B0C0D0E0F10111213"
+	spec := []string{"%v", "%#v", "%t", "%b", "%c", "%d", "%o", "%O", "%x", "%X", "%U", "%e", "%E", "%f", "%F", "%g", "%G"}
+	for _, v := range spec {
+		s.Require().Equal(exp, fmt.Sprintf(v, myAddr), v)
+	}
+}
+
+func (s *addressTestSuite) TestUnmarshalJSONConsAddressFailed() {
+	addr := &types.ConsAddress{}
+	err := addr.UnmarshalJSON(nil)
+	s.Require().Error(err)
+}
+
+func (s *addressTestSuite) TestUnmarshalJSONConsAddressWithEmptyString() {
+	addr := &types.ConsAddress{}
+	err := addr.UnmarshalJSON([]byte{34, 34})
+	s.Require().NoError(err)
+	s.Require().Equal(&types.ConsAddress{}, addr)
+}
+
+func (s *addressTestSuite) TestUnmarshalYAMLConsAddressFailed() {
+	malformedYAML := []byte("k:k:K:")
+	addr := &types.ConsAddress{}
+	err := addr.UnmarshalYAML(malformedYAML)
+	s.Require().Error(err)
+}
+
+func (s *addressTestSuite) TestUnmarshalYAMLConsAddressWithEmptyString() {
+	addr := &types.ConsAddress{}
+	err := addr.UnmarshalYAML([]byte{34, 34})
+	s.Require().NoError(err)
+	s.Require().Equal(&types.ConsAddress{}, addr)
+}
+
+func (s *addressTestSuite) TestFormatConsAddressAsString() {
+	consAddr := types.ConsAddress([]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+	actual := fmt.Sprintf("%s", consAddr)
+
+	hrp := types.GetConfig().GetBech32ConsensusAddrPrefix()
+	expected, err := bech32.ConvertAndEncode(hrp, consAddr)
```

Comment:
Why not use the string value here instead of converting. It helps with regression is much more readable to humans as well.

Reply:
Done